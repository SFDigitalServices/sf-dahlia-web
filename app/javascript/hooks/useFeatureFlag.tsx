import { useFlag as useFlagUnleash, useFlagsStatus } from "@unleash/proxy-client-react"

// If you want to prevent a user from changing this feature flag via the URL, you can add it to the urlBlockList set.
// When set, the feature flag will only be determined by the Unleash API.
const urlBlockList = new Set([""])

export const useFeatureFlag = (flagName: string, defaultValue: boolean) => {
  const urlParams = new URLSearchParams(window.location.search)
  const flagSearchParam = `featureFlag[${flagName}]`

  const doesURLHaveFlag = urlParams.has(flagSearchParam)
  const flagFromUrl = urlParams.get(flagSearchParam)

  const { flagsError, flagsReady } = useFlagsStatus()

  const unleashFlag = useFlagUnleash(flagName)

  if (doesURLHaveFlag && !urlBlockList.has(flagName) && process.env.UNLEASH_ENV === "development") {
    if (flagFromUrl === "true") {
      return { flagsReady: true, unleashFlag: true }
    } else if (flagFromUrl === "false") {
      return { flagsReady: true, unleashFlag: false }
    }
  }

  if (flagsError || unleashFlag === undefined) {
    console.error(flagsError)
    return { flagsReady, unleashFlag: defaultValue }
  } else {
    return { flagsReady, unleashFlag }
  }
}
