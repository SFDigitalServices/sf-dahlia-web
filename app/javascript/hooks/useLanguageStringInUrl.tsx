import { useEffect, useState } from "react"
import { getCurrentLanguage } from "../util/languageUtil"

const useLanguageStringInUrl = () => {
  const [languageFromStringInUrl, setLanguageFromStringInUrl] = useState("en")
  const lang = getCurrentLanguage()

  useEffect(() => {
    setLanguageFromStringInUrl(lang)
  }, [lang])

  return languageFromStringInUrl
}

export default useLanguageStringInUrl
