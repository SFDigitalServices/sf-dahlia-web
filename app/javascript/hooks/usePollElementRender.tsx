import { useState, useEffect } from "react"

const usePollElementRender = (selector, disable) => {
  const [elementHasBeenRenderedStatus, setElementHasBeenRenderedStatus] = useState(false)

  useEffect(() => {
    if (disable) {
      setElementHasBeenRenderedStatus(false)
      return () => elementHasBeenRenderedStatus
    }
    /*
      This useEffect is checking for when an element get added to the dom
     */
    let iterations = 0

    /*
     * Poll every 1/3 second for 30 seconds checking for google translate to add language dropdown
     */
    const interval = setInterval(() => {
      iterations++
      const elementInDom = document.querySelector(selector)
      if (elementInDom && !elementHasBeenRenderedStatus) {
        setElementHasBeenRenderedStatus(true)
        clearInterval(interval)
      }
      if (iterations > 90) {
        clearInterval(interval)
      }
    }, 300)

    return () => clearInterval(interval)

    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  return elementHasBeenRenderedStatus
}

export default usePollElementRender
