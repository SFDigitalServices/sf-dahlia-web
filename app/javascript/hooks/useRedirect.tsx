import { useState, useEffect } from "react"
import { clearSiteAlertMessage, setSiteAlertMessage } from "../components/SiteAlert"
import { t } from "@bloom-housing/ui-components"

const handleRedirect = (path: string) => {
  const isRedirectSet = sessionStorage.getItem("redirect")
  if (!isRedirectSet) {
    sessionStorage.setItem("redirect", path)

    setSiteAlertMessage(t("signIn.loginRequired"), "secondary")
  } else {
    sessionStorage.removeItem("redirect")
    clearSiteAlertMessage("secondary")
  }
}

const useRedirect = () => {
  const [redirect, setRedirect] = useState<string | null>(null)

  useEffect(() => {
    const redirect = sessionStorage.getItem("redirect")
    sessionStorage.removeItem("redirect")
    if (redirect) {
      setRedirect(redirect)
    } else {
      setRedirect(null)
      clearSiteAlertMessage("secondary")
    }
  }, [])

  return { redirect, handleRedirect }
}

export default useRedirect
