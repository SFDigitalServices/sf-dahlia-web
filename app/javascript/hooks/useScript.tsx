/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/no-unsafe-argument */
import { useState, useEffect } from "react"

const useScript = (src) => {
  // Keep track of script status ("idle", "loading", "ready", "error")
  const [status, setStatus] = useState(src ? "loading" : "idle")
  useEffect(
    () => {
      // Allow falsy src value if waiting on other data needed for
      // constructing the script URL passed to this hook.
      if (!src) {
        setStatus("idle")
        return
      }
      // Fetch existing script element by src
      // It may have been added by another intance of this hook
      // TODO(DAH-1581): Remove any type on line 18
      let script: any = document.querySelector(`script[src="${src}"]`)
      if (!script) {
        // Create script
        script = document.createElement("script")
        script.src = src
        script.async = true
        // eslint-disable-next-line unicorn/prefer-dom-node-dataset
        script.setAttribute("data-status", "loading")
        // Add script to document body
        document.body.append(script)
        // Store status in attribute on script
        // This can be read by other instances of this hook
        const setAttributeFromEvent = (event) => {
          // eslint-disable-next-line unicorn/prefer-dom-node-dataset
          script.setAttribute("data-status", event.type === "load" ? "ready" : "error")
        }
        script.addEventListener("load", setAttributeFromEvent)
        script.addEventListener("error", setAttributeFromEvent)
      } else {
        // Grab existing script status from attribute and set to state.
        // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
        setStatus(script.getAttribute("data-status"))
      }
      // Script event handler to update status in state
      // Note: Even if the script already exists we still need to add
      // event handlers to update the state for *this* hook instance.
      const setStateFromEvent = (event) => {
        setStatus(event.type === "load" ? "ready" : "error")
      }
      // Add event listeners
      script.addEventListener("load", setStateFromEvent)
      script.addEventListener("error", setStateFromEvent)
      // Remove event listeners on cleanup
      return () => {
        if (script) {
          script.removeEventListener("load", setStateFromEvent)
          script.removeEventListener("error", setStateFromEvent)
        }
      }
    },
    [src] // Only re-run effect if script src changes
  )
  return status
}
export default useScript
