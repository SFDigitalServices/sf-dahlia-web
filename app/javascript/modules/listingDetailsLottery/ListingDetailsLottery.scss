.modal {
  @apply max-w-0;
}

.modal__inner {
  @apply p-0;
}

.modal__close {
  z-index: 100;
}
