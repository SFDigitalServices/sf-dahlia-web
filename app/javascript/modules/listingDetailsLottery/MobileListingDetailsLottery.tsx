import React from "react"
import { ListingDetailItem, Mobile, t } from "@bloom-housing/ui-components"
import { RailsListing } from "../listings/SharedHelpers"
import { ListingDetailsLotteryResults } from "./ListingDetailsLotteryResults"
import { ListingDetailsProcess } from "../listingDetailsAside/ListingDetailsProcess"
import { isOpen } from "../../util/listingUtil"
import { ListingDetailsLotteryInfo } from "./LotteryDetailsLotteryInfo"
import { ListingDetailsWaitlist } from "../listingDetailsAside/ListingDetailsWaitlist"
import { ListingDetailsOpenHouses } from "../listingDetailsAside/ListingDetailsOpenHouses"

export interface ListingDetailsLotteryProps {
  imageSrc: string
  listing: RailsListing
  whatToExpectContent: string
}

export const MobileListingDetailsLottery = ({ imageSrc, listing, whatToExpectContent }: ListingDetailsLotteryProps) => {
  const isApplicationOpen = isOpen(listing)

  return (
    listing &&
    !isApplicationOpen && (
      <Mobile>
        <ListingDetailItem
          imageAlt={""}
          imageSrc={imageSrc}
          title={t("lottery")}
          subtitle={t("lottery.lotteryInfoSubheader")}
        >
          <ListingDetailsLotteryInfo listing={listing} />
          <ListingDetailsLotteryResults listing={listing} />
          <ListingDetailsWaitlist listing={listing} />
          <ListingDetailsOpenHouses listing={listing} />
          <ListingDetailsProcess listing={listing} isApplicationOpen={isApplicationOpen} whatToExpectContent={whatToExpectContent} />
        </ListingDetailItem>
      </Mobile>
    )
  )
}
