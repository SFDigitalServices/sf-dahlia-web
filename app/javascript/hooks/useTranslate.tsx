/*
 * This hook is for using an older version of Google Translate
 * that requires work to ensure the timing of script loading and event
 * dispatching is correct. Essentially, we inject a Google Translate script that adds
 * a <select> element to the dom. That <select> element is a dropdown of different languages
 * that we can machine translate the page with. We hide that <select> with css.
 * Once that <select> element is rendered (even though hidden), we use the language
 * set in the url to dispatch an event setting that <select>'s <option>
 * to the corresponding language. This gives an experience to the end user that
 * the page automatically got translated.
 * Soon to be deprecated
 */

import { useEffect } from "react"
import usePollElementRender from "./usePollElementRender"
import { getCurrentLanguage } from "../util/languageUtil"
import useScript from "./useScript"
// eslint-disable-next-line @typescript-eslint/no-explicit-any
const tWindow = window as any
const languageMap = {
  zh: "zh-TW",
}

const initGoogleTranslate = () => {
  // eslint-disable-next-line no-new
  new tWindow.google.translate.TranslateElement(
    {
      pageLanguage: "en",
      includedLanguages: "en,es,tl,zh-TW",
      layout: tWindow.google.translate.TranslateElement.InlineLayout.HORIZONTAL,
      autoDisplay: false,
      multilanguagePage: true,
    },
    "google_translate_element"
  )
}

tWindow.initGoogleTranslate = initGoogleTranslate

const useTranslate = (disable) => {
  const languageInRoute = getCurrentLanguage()
  useScript("//translate.google.com/translate_a/element.js?cb=initGoogleTranslate", disable)
  /*
   * It seems the <select> and the <options> get added at different times.
   * We want to target the <select> for dispatching the change event, but we have to wait
   * until the <options> appear to dispatch the event.
   */
  const googleTranslateDropdownElHasBeenRendered = usePollElementRender(
    "select.goog-te-combo option",
    disable
  )

  useEffect(() => {
    if (!disable && languageInRoute && googleTranslateDropdownElHasBeenRendered) {
      // TODO(DAH-1581): Remove any type on line 55
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      const selectInDom: any = document.querySelector("select.goog-te-combo")
      selectInDom.value = languageMap[languageInRoute] || languageInRoute
      const ev = new Event("change", { bubbles: true })
      selectInDom.dispatchEvent(ev)
    }
  }, [languageInRoute, googleTranslateDropdownElHasBeenRendered, disable])
}

export default useTranslate
