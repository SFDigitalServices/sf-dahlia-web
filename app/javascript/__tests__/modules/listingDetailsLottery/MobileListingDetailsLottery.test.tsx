import React from "react"
import { render } from "@testing-library/react"
import { MobileListingDetailsLottery } from "../../../modules/listingDetailsLottery/MobileListingDetailsLottery"
import { fcfsSaleListing } from "../../data/RailsSaleListing/listing-sale-fcfs"
import { closedRentalListing } from "../../data/RailsRentalListing/listing-rental-closed"
import { useFeatureFlag } from "../../../hooks/useFeatureFlag"
import { notYetOpenSaleFcfsListing } from "../../data/RailsSaleListing/listing-sale-fcfs-not-yet-open"
import { openSaleListing } from "../../data/RailsSaleListing/listing-sale-open"

jest.mock("../../../hooks/useFeatureFlag", () => ({
  useFeatureFlag: jest.fn(),
}))

beforeEach(() => {
  window.matchMedia = jest.fn().mockImplementation((query) => {
    return {
      matches: true,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    }
  })
})

describe("MobileListingDetailsLottery", () => {
  it("does not display when fcfs sale listing", () => {
    ;(useFeatureFlag as jest.Mock).mockReturnValue({ flagsReady: true, unleashFlag: true })

    const { asFragment } = render(
      <MobileListingDetailsLottery imageSrc="" listing={fcfsSaleListing} />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it("does display when fcfs toggle is disabled", () => {
    ;(useFeatureFlag as jest.Mock).mockReturnValue({ flagsReady: true, unleashFlag: false })

    const { asFragment } = render(
      <MobileListingDetailsLottery imageSrc="" listing={notYetOpenSaleFcfsListing} />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it("does display when lottery listing is closed", () => {
    ;(useFeatureFlag as jest.Mock).mockReturnValue({ flagsReady: true, unleashFlag: true })

    const { asFragment } = render(
      <MobileListingDetailsLottery listing={closedRentalListing} imageSrc={""} />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it("does not display if listing is open", () => {
    ;(useFeatureFlag as jest.Mock).mockReturnValue({ flagsReady: true, unleashFlag: true })

    const { asFragment } = render(
      <MobileListingDetailsLottery listing={openSaleListing} imageSrc={""} />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
