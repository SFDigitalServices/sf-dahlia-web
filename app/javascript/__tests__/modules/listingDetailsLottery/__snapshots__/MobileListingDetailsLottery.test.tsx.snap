// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileListingDetailsLottery does display when fcfs toggle is disabled 1`] = `
<DocumentFragment>
  <li
    class="responsive-content-item "
  >
    <header
      class="detail-header "
    >
      <span
        class="detail-header__image-container"
      >
        <img
          alt=""
          class="detail-header__image "
          src=""
        />
      </span>
      <hgroup
        class="detail-header__hgroup"
      >
        <h2
          class="detail-header__title"
        >
          Lottery
        </h2>
        <span
          class="detail-header__subtitle"
        >
          Lottery selection, important dates and contact
        </span>
        <span
          class="ui-icon ui-medium"
        >
          <svg
            fill="currentColor"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </hgroup>
    </header>
    <div
      class="border-b border-gray-400 md:border-b-0"
    >
      <section
        class="aside-block "
      >
        <h2
          class="text__underline-weighted"
        >
          Lottery
        </h2>
        <div
          class="text-sm text-gray-750"
        >
          <p
            class="flex justify-between mb-4"
          >
            <span>
              January 17, 2021
            </span>
            <span
              class="font-bold"
            >
              7:00 pm
            </span>
          </p>
          <div
            class="text-gray-700"
          >
            <p>
              MOHCD Offices
            </p>
          </div>
          <p
            class="mt-4 text-gray-700"
          >
            An open lottery will be held to draw numbers to determine selection. Attendance is optional.
          </p>
        </div>
      </section>
    </div>
    <div
      class="border-b border-gray-400 md:border-b-0"
    >
      <section
        class="aside-block "
      >
        <h2
          class="text__underline-weighted"
        >
          Lottery Results
        </h2>
        <div
          class="text-sm text-gray-750"
        >
          <p
            class="mb-4"
          >
            January 24, 2021
          </p>
          <p
            class="text-gray-700"
          >
            Complete lottery results will be posted on this date by 5 pm.
          </p>
        </div>
      </section>
    </div>
    <div
      class="border-b border-gray-400 md:border-b-0"
    >
      <section
        class="aside-block"
      >
        <h2
          class="text__underline-weighted"
        >
          What to Expect
        </h2>
        <div
          class="text-sm text-gray-750"
        >
          <span>
            Applicants will be contacted by the agent in lottery rank order until vacancies are filled.
          </span>
          <div
            class="mt-2"
          >
            <div>
              <button
                aria-expanded="false"
                aria-label="Show more about what to expect"
                class="button is-unstyled m-0 no-underline has-toggle"
                type="button"
              >
                Show more
              </button>
            </div>
          </div>
        </div>
      </section>
    </div>
    <div
      class="border-b border-gray-400 md:border-b-0 last:border-b-0"
    >
      <section
        class="aside-block"
      >
        <h2
          class="text__underline-weighted"
        >
          Contact
        </h2>
        <p
          class="text-xl"
        >
          Matthew Burns
        </p>
        <p
          class="text-gray-700 translate"
        >
          Leasing Agent
        </p>
        <p
          class="mt-3"
        >
          <a
            href="tel:Call 415 5552323"
          >
            <span
              class="ui-icon ui-medium pr-2"
            >
              <svg
                fill="#0077DA"
                viewBox="0 0 32 32"
              >
                <path
                  d="M22 20c-2 2-2 4-4 4s-4-2-6-4-4-4-4-6 2-2 4-4-4-8-6-8-6 6-6 6c0 4 4.109 12.109 8 16s12 8 16 8c0 0 6-4 6-6s-6-8-8-6z"
                />
              </svg>
            </span>
            Call (415) 555-2323
          </a>
        </p>
        <p
          class="text-xs text-gray-700"
        >
          Due to high call volume you may hear a message.
        </p>
        <p
          class="my-3"
        >
          <a
            href="mailto:ae@email.com"
          >
            <span
              class="ui-icon ui-medium pr-2"
            >
              <svg
                fill="#0077DA"
                viewBox="0 0 32 32"
              >
                <path
                  d="M29.091 27.637h-26.182c-1.606 0-2.909-1.304-2.909-2.909v-17.454c0-1.606 1.304-2.909 2.909-2.909h26.182c1.606 0 2.909 1.304 2.909 2.909v17.454c0 1.606-1.304 2.909-2.909 2.909zM27.637 24.728l-8.338-7.296-3.299 2.933-3.299-2.933-8.338 7.296h23.272zM2.909 23.272l8.182-7.272-8.182-7.272v14.546zM4.363 7.272l11.637 10.182 11.637-10.182h-23.272zM29.091 8.728l-8.182 7.272 8.182 7.272v-14.546z"
                />
              </svg>
            </span>
            Email
          </a>
        </p>
        <p
          class="text-gray-700 mb-1"
        >
          <span>
            1 South Van Ness Ave  
            <br />
            San Francisco, CA 94131
          </span>
        </p>
        <p
          class="mb-4"
        >
          <a
            class="inline-block pt-1"
            href="https://www.google.com/maps/place/1 South Van Ness Ave , San Francisco, CA 94131"
            rel="noreferrer noopener"
            target="_blank"
          >
            <span
              class="ui-icon ui-medium pr-2"
            >
              <svg
                fill="#0077DA"
                viewBox="0 0 18 32"
              >
                <path
                  d="M13.714 11.429q0-1.893-1.339-3.232t-3.232-1.339-3.232 1.339-1.339 3.232 1.339 3.232 3.232 1.339 3.232-1.339 1.339-3.232zM18.286 11.429q0 1.947-0.589 3.197l-6.501 13.821q-0.286 0.589-0.848 0.928t-1.205 0.339-1.205-0.339-0.83-0.928l-6.518-13.821q-0.589-1.25-0.589-3.197 0-3.786 2.678-6.464t6.464-2.678 6.464 2.678 2.678 6.464z"
                />
              </svg>
            </span>
            Get Directions
          </a>
        </p>
        <div
          class="my-3"
        >
          <h3
            class="text__caps-weighted"
          >
            Office Hours
          </h3>
          <div
            class="text-gray-800 text-sm markdown"
          >
            <span
              class="translate"
            >
              <span>
                8-4 Mon-Thurs

                <br />
                9-4 Fri

                <br />
                10-2 Weekends
              </span>
            </span>
          </div>
        </div>
      </section>
    </div>
    <div
      class="border-b border-gray-400 md:border-b-0 last:border-b-0"
    >
      <section
        class="aside-block "
      >
        <h2
          class="text__underline-weighted"
        >
          Housing Program
        </h2>
        <div
          class="text-sm text-gray-750"
        >
          <a
            class="text-base"
            href="https://sfmohcd.org/for-buyers"
            target="_blank"
          >
            Below Market Rate
          </a>
        </div>
      </section>
    </div>
    <div
      class="border-b border-gray-400 md:border-b-0 last:border-b-0"
    >
      <section
        class="aside-block "
      >
        <div
          class="text-sm text-gray-750"
        >
          <p>
            Listing Updated: April 11, 2022
          </p>
          <p
            class="mt-1"
          >
            <a
              class=""
              href="https://thefrontsteps.com/featured-listings/sfar/mls-481663-2820_scott_street_san_francisco_ca_94123/"
              target="_blank"
            >
              See this unit on MLS
            </a>
          </p>
        </div>
      </section>
    </div>
  </li>
</DocumentFragment>
`;

exports[`MobileListingDetailsLottery does display when lottery listing is closed 1`] = `
<DocumentFragment>
  <li
    class="responsive-content-item "
  >
    <header
      class="detail-header "
    >
      <span
        class="detail-header__image-container"
      >
        <img
          alt=""
          class="detail-header__image "
          src=""
        />
      </span>
      <hgroup
        class="detail-header__hgroup"
      >
        <h2
          class="detail-header__title"
        >
          Lottery
        </h2>
        <span
          class="detail-header__subtitle"
        >
          Lottery selection, important dates and contact
        </span>
        <span
          class="ui-icon ui-medium"
        >
          <svg
            fill="currentColor"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.267 6.133l-12.267 12.267-12.267-12.267-3.733 3.733 16 16 16-16z"
            />
          </svg>
        </span>
      </hgroup>
    </header>
    <div
      class="border-b border-gray-400 md:border-b-0"
    >
      <section
        class="aside-block "
      >
        <h2
          class="text__underline-weighted"
        >
          Lottery
        </h2>
        <div
          class="text-sm text-gray-750"
        >
          <p
            class="flex justify-between mb-4"
          >
            <span>
              March 22, 2023
            </span>
            <span
              class="font-bold"
            >
              6:00 pm
            </span>
          </p>
          <div
            class="text-gray-700"
          >
            <div>
              <p>
                The lottery will not be in person, but instead will be a virtual lottery.
              </p>
              <p>
                Â 
              </p>
              <p>
                Come back to this page on the day of the lottery for a link you can click on to watch the lottery live.
              </p>
              <p>
                <br />
              </p>
              <p>
                <br />
              </p>
            </div>
          </div>
          <p
            class="mt-4 text-gray-700"
          >
            An open lottery will be held to draw numbers to determine selection. Attendance is optional.
          </p>
        </div>
      </section>
    </div>
    <div
      class="border-b border-gray-400 md:border-b-0"
    >
      <section
        class="aside-block "
      >
        <h2
          class="text__underline-weighted"
        >
          Lottery Results
        </h2>
        <div
          class="text-sm text-gray-750"
        >
          <p
            class="mb-4"
          >
            March 23, 2023
          </p>
          <p
            class="text-gray-700"
          >
            Complete lottery results will be posted on this date by 5 pm.
          </p>
        </div>
      </section>
    </div>
    <div
      class="open-houses border-b border-gray-400 md:border-b-0"
    >
      <section
        class="aside-block"
      >
        <h4
          class="text__underline-weighted"
        >
          Open Houses
        </h4>
        <div
          class="false"
        >
          <p
            class="text text-gray-800 pb-2 flex justify-between items-center"
          >
            <span
              class="inline-block text-sm uppercase normal-case"
            >
              February 23, 2018
            </span>
            <span
              class="inline-block text-xs font-bold ml-5 font-alt-sans"
            >
              9:00AM - 12:00PM
            </span>
          </p>
          <div
            class="text-sm text-gray-700 false"
          >
            <div
              class="flex flex-col"
            >
              <span
                class="links-space translate"
              >
                <span>
                  Library
                </span>
              </span>
              <span>
                123 Van ness, San Francisco
              </span>
            </div>
          </div>
        </div>
      </section>
    </div>
    <div
      class="border-b border-gray-400 md:border-b-0"
    >
      <section
        class="aside-block"
      >
        <h2
          class="text__underline-weighted"
        >
          What to Expect
        </h2>
        <div
          class="text-sm text-gray-750"
        >
          <span>
            Applicants will be contacted by the agent in lottery rank order until vacancies are filled.
          </span>
          <div
            class="mt-2"
          >
            <div>
              <button
                aria-expanded="false"
                aria-label="Show more about what to expect"
                class="button is-unstyled m-0 no-underline has-toggle"
                type="button"
              >
                Show more
              </button>
            </div>
          </div>
        </div>
      </section>
    </div>
    <div
      class="border-b border-gray-400 md:border-b-0 last:border-b-0"
    >
      <section
        class="aside-block "
      >
        <div
          class="text-sm text-gray-750"
        >
          <p>
            Listing Updated: April 1, 2022
          </p>
        </div>
      </section>
    </div>
  </li>
</DocumentFragment>
`;

exports[`MobileListingDetailsLottery does not display when fcfs sale listing 1`] = `<DocumentFragment />`;
